runtime/compiler/z/codegen/J9TreeEvaluator.hpp

```diff
@@ -90,8 +90,8 @@ class OMR_EXTENSIBLE TreeEvaluator: public J9::TreeEvaluator
    {
    public:
 
-   static TR::Register *awrtbarEvaluator(TR::Node *node, TR::CodeGenerator *cg);
-   static TR::Register *awrtbariEvaluator(TR::Node *node, TR::CodeGenerator *cg);
+   static TR::Register *writeBarrierEvaluator(TR::Node *node, TR::CodeGenerator *cg);
+   static TR::Register *writeBarrierIndirectEvaluator(TR::Node *node, TR::CodeGenerator *cg);
```

Comment:
Why did we change these names? Canonical naming convention for evaluators is the name of the IL followed by "Evaluator". It makes searching the codebase very easy if everything is consistent.

Reply:
Also can these evaluator definitions be grouped down below with all the other ones? No reason for them to be separated. Do you agree?

Reply:
> Why did we change these names? Canonical naming convention for evaluators is the name of the IL followed by "Evaluator". It makes searching the codebase very easy if everything is consistent.

So as it stands currently we hack the TreeEvaluatorTable to add the `awrtbar*` evaluator names in:
https://github.com/eclipse/openj9/blob/4577abf9392aecd6574ca4a5a84c6761894123d2/runtime/compiler/z/codegen/J9TreeEvaluator.cpp#L974-L979

With this pull request we have proper implementations for all read and write barrier opcodes with field watch support. So the old `awrtBar*` evaluators need to be renamed because there are new `awrtbar*` evaluators. I removed these entries from `TEMPORARY_initJ9S390TreeEvaluatorTable` (because we can use the default names now) and renamed them accordingly so that the new `awrtbar*` evaluators can call the appropriate "old" routines after managing the field watch work. 

Perhaps the name here can be `writeBarrierEvaluatorHelper` instead of `writeBarrierEvaluator`? That would better indicate the purpose of the routine. I can also squash the two routines together if you think that's better. I prefer the first alternative though. Do you have any preference?

Reply:
I'm not sure I'm following the logic for the renaming. Why can't we keep the names as is and inline the field watch logic right in `awrtbarEvaluator` and `awrtbariEvaluator`?

Reply:
I'm also fairly sure the hack you noted above is not needed. The evaluator for `awrtbar` for example in OMR can be found in [1]. With the change from your dependent OMR PR eclipse/omr#3571 that value will change to `TR::TreeEvaluator::awrtbarEvaluator` and the IPATH of OpenJ9 will make that resolve to the OpenJ9 version.

The OpenJ9 `TR::TreeEvaluator::awrtbarEvaluator` should be the existing evaluator for write barriers which should have just an `if` at the start to see if we need to generate the field watch code as well.

[1] https://github.com/eclipse/omr/blob/4edb1472c52c529d1e5343e9a7332164b9fda891/compiler/z/codegen/TreeEvaluatorTable.hpp#L74
[2] https://github.com/eclipse/omr/pull/3571/files#diff-4006a8320b0f831f81d4b9ffec05735eR74

Reply:
Yes, after the OMR change is merged the hack won't be needed. I was just mentioning why it was there initially.

I've now consolidated the  `awrtbar` evaluators have in z/codegen and field watch support is inlined there. The relevant commits are: 
a7c61d2 + c77e980